### PR TITLE
Adding cluster config to config number of concurrent tasks per instance for minion task: SegmentGenerationAndPushTaskGenerator

### PR DESCRIPTION
**Pull request by fx19880617**
## Description

- Adding cluster config: `SegmentGenerationAndPushTask.numConcurrentTasksPerInstance` to config number of concurrent tasks per instance for minion task: SegmentGenerationAndPushTaskGenerator
- Adding new API in ClusterInfoAccessor to fetch cluster config

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java [hunk 1]
```diff
@@ -157,4 +160,20 @@ public void setRealtimeToOfflineSegmentsTaskMetadata(
   public String getVipUrl() {
     return _controllerConf.generateVipUrl();
   }
+
+  /**
+   * Get the cluster config for a given cluster config.
```
Comment (icefury71) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java:165]: nit: "For a given config name"

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/SegmentGenerationAndPushTaskGenerator.java [hunk 2]
```diff
@@ -103,6 +104,21 @@ public String getTaskType() {
     return MinionConstants.SegmentGenerationAndPushTask.TASK_TYPE;
   }
 
+  @Override
+  public int getNumConcurrentTasksPerInstance() {
+    String numConcurrentTasksPerInstance = _clusterInfoAccessor
```
Comment (icefury71) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/SegmentGenerationAndPushTaskGenerator.java:109]: nit: numConcurrentTasksPerInstanceStr

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/SegmentGenerationAndPushTaskGenerator.java [hunk 3]
```diff
@@ -103,6 +104,21 @@ public String getTaskType() {
...
+        .getClusterConfig(MinionConstants.SegmentGenerationAndPushTask.CONFIG_NUMBER_CONCURRENT_TASKS_PER_INSTANCE);
```
Comment (icefury71) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/SegmentGenerationAndPushTaskGenerator.java:110]: maybe use static import ?

**Review by icefury71 — APPROVED** [review]

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java (hunk 1 above)
Comment (fx19880617) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java]: done

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/SegmentGenerationAndPushTaskGenerator.java (hunk 3 above)
Comment (fx19880617) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/generator/SegmentGenerationAndPushTaskGenerator.java:110]: yeah, I just keep the same convention as in this file. Maybe good to have another PR to swipe all the conventions ;)

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java [hunk 4]
```diff
@@ -157,4 +160,20 @@ public void setRealtimeToOfflineSegmentsTaskMetadata(
...
+   */
+  public String getClusterConfig(String configName) {
+    HelixConfigScope helixConfigScope = new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER)
+        .forCluster(_pinotHelixResourceManager.getHelixClusterName()).build();
+    Map<String, String> configMap =
+        _pinotHelixResourceManager.getHelixAdmin().getConfig(helixConfigScope, Arrays.asList(configName));
+    if (configMap == null || !configMap.containsKey(configName)) {
+      return null;
+    }
+    return configMap.get(configName);
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java:177]: (nit) Can be simplified to
```suggestion
    return configMap != null ? configMap.get(configName) : null;
```

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java [hunk 5]
```diff
@@ -157,4 +160,20 @@ public void setRealtimeToOfflineSegmentsTaskMetadata(
...
+  /**
+   * Get the cluster config for a given config name, return null if not found.
+   *
+   * @return cluster config
+   */
+  public String getClusterConfig(String configName) {
+    HelixConfigScope helixConfigScope = new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.CLUSTER)
+        .forCluster(_pinotHelixResourceManager.getHelixClusterName()).build();
+    Map<String, String> configMap =
+        _pinotHelixResourceManager.getHelixAdmin().getConfig(helixConfigScope, Arrays.asList(configName));
```
Comment (Jackie-Jiang) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/ClusterInfoAccessor.java:173]: (nit)
```suggestion
        _pinotHelixResourceManager.getHelixAdmin().getConfig(helixConfigScope, Collections.singletonList(configName));
```

File: pinot-core/src/main/java/org/apache/pinot/core/common/MinionConstants.java [hunk 6]
```diff
@@ -92,6 +92,8 @@ private MinionConstants() {
   // Generate segment and push to controller based on batch ingestion configs
   public static class SegmentGenerationAndPushTask {
     public static final String TASK_TYPE = "SegmentGenerationAndPushTask";
+    public static final String CONFIG_NUMBER_CONCURRENT_TASKS_PER_INSTANCE =
+        "pinot.minion.task.generator.SegmentGenerationAndPushTask.numConcurrentTasksPerInstance";
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/common/MinionConstants.java:96]: Simplify this key to `SegmentGenerationAndPushTask.numConcurrentTasksPerInstance`? The current one looks like an instance config key

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise

**Pull request merged**